grakn-graph/src/main/java/ai/grakn/graph/internal/InstanceImpl.java

```diff
@@ -88,26 +92,20 @@
      */
     @Override
     public void delete() {
-        InstanceImpl<?, ?> parent = this;
-        Set<CastingImpl> castings = parent.castings();
+        Set<Relation> relations = castingsInstance().map(Casting::getRelation).collect(Collectors.toSet());
```

Comment:
Nope, wrong. There is a line below this and before actually iterating over the set that deletes the instance and therefore eliminates the ability to get the relations. 

It needs to be done in this wonky order otherwise we get null pointers when we empty relations and auto delete them.